module/Finna/src/Finna/Recommend/OntologyDeferred.php

```diff
@@ -0,0 +1,144 @@
+<?php
```

Comment:
Deferred-toimintamallin tulisi olla aina käytössä, kun on ulkoisia riippuvuuksia, jolloin voidaan jättää tämä pois ja tehdä vain AJAX-toiminnallisuudella Ontology-recommenderissa.

Reply:
Onko tällaisesta jotain olemassa olevaa esimerkkiä? Ontology-recommender pitäisi ilmeisesti jotenkin muokata tukemaan kahta templatea tai sitten templatessa päätellä kummasta kutsusta kyse? Lopputuloksena on kai joka tapauksessa "normaalista" poikkeavalla tavalla toimiva recommend-moduuli joka on oikeasti deferred, mutta ei nimetty sellaiseksi?

Reply:
Ei siinä ole mitään epänormaalia, että recommender tekee asiat niinkuin sitä sattuu huvittamaan. Deferred-versiot ovat olemassa erotuksena sellaisista suosittelijoista, joista on myös ei-Deferred-versiot. Esim. SideFacets oli alunperin vain ei-Deferred, mutta haluttiin saada hakutulokset nopeammin kuin fasetointi tapahtuu, joten tehtiin Deferred-versio.

Reply:
Joo, noin päin on helpompi tehdä, että lisää deferred-version. Mutta jos ainoa toteutus on deferred, niin helperin pitäisi päätellä onko kyse normaalista vai AJAX-kutsusta. Onko tällaisesta valmista esimerkkiä, vai kehitänkö oman tavan?